jpa/keyset-pagination/boot-data-window-pagination/src/main/java/com/example/keysetpagination/services/AnimalService.java

```diff
@@ -60,22 +65,33 @@ private Pageable createPageable(FindAnimalsQuery findAnimalsQuery) {
         return PageRequest.of(pageNo, findAnimalsQuery.pageSize(), sort);
     }
 
-    public Window<AnimalResponse> searchAnimals(List<SearchCriteria> searchCriteriaList, int pageSize, Long scrollId) {
+    public Window<AnimalResponse> searchAnimals(SearchRequest searchRequest, int pageSize, Long scrollId) {
 
         Specification<Animal> specification =
-                animalEntitySpecification.specificationBuilder(searchCriteriaList, Animal.class);
+                animalEntitySpecification.specificationBuilder(searchRequest.getSearchCriteriaList(), Animal.class);
 
         // Create initial ScrollPosition or continue from the given scrollId
         ScrollPosition position = scrollId == null
                 ? ScrollPosition.keyset()
                 : ScrollPosition.of(Collections.singletonMap("id", scrollId), ScrollPosition.Direction.FORWARD);
 
+        // Parse and create sort orders
+        List<Sort.Order> orders = CollectionUtils.isEmpty(searchRequest.getSortRequests())
+                ? Collections.singletonList(new Sort.Order(Sort.Direction.ASC, "id"))
+                : searchRequest.getSortRequests().stream()
+                        .map(sortRequest -> {
+                            Sort.Direction direction = "desc"
+                                            .equalsIgnoreCase(Optional.ofNullable(sortRequest.getDirection())
+                                                    .orElse("asc"))
+                                    ? Sort.Direction.DESC
+                                    : Sort.Direction.ASC;
+                            return new Sort.Order(direction, sortRequest.getField());
+                        })
+                        .toList();
+
+        log.debug("Executing search with specification: {} and sort orders: {}", specification, orders);
```

Comment:
@coderabbitai, how to print the specification here in human readable format?

Reply:
@coderabbitai, can you please help me to write custom method to traverse and display the specification components.

Reply:
@coderabbitai, can you please create an issue to implement above feature